From 2caae188279acd40bbd19c92702238709917fa48 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Thu, 6 Apr 2017 11:37:55 +0000
Subject: [PATCH 3/4] EDAC, thunderx: Change LMC index calculation

Git-commit: 3d2d8c0f84ca518da9210b3f855643efd1629e07
Patch-mainline: v4.12-rc1
References: bsc#1035479

Shift the node number by 3 bits instead of 8 allowing proper functioning
with default EDAC_MAX_MCS.

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Cc: David Daney <david.daney@cavium.com>
Cc: Jan.Glauber@cavium.com
Cc: linux-edac <linux-edac@vger.kernel.org>
Link: http://lkml.kernel.org/r/20170406113755.17082-1-s.temerkhanov@gmail.com
Signed-off-by: Borislav Petkov <bp@suse.de>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/edac/thunderx_edac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/edac/thunderx_edac.c b/drivers/edac/thunderx_edac.c
index f39f4bb..955f73c 100644
--- a/drivers/edac/thunderx_edac.c
+++ b/drivers/edac/thunderx_edac.c
@@ -670,7 +670,7 @@ static inline int pci_dev_to_mc_idx(struct pci_dev *pdev)
 	int node = dev_to_node(&pdev->dev);
 	int ret = PCI_FUNC(pdev->devfn);
 
-	ret += max(node, 0) << 8;
+	ret += max(node, 0) << 3;
 
 	return ret;
 }
-- 
2.11.0

